### PR TITLE
Fixed overflow error & made left/right available while custom menu open

### PR DESCRIPTION
**Pull request by RensDofferhoff**
Fixed overflow error on Ribbon button enter & made left/right available while custom menu open.

### CODE REVIEW

File: Desktop/components/JASP/Widgets/CustomMenu.qml [hunk 1]
```diff
@@ -45,22 +45,28 @@ FocusScope
 
 	property int	currentIndex: -1
 
+	onSourceItemChanged: { menu.currentIndex = -1; }
+
 	Keys.onPressed: (event)=>
 	{
 		if (event.key === Qt.Key_Up || event.key === Qt.Key_Backtab)
```
Comment (JorisGoosen) [inline Desktop/components/JASP/Widgets/CustomMenu.qml]: Can you make this a switch?

Comment (RensDofferhoff) [reply Desktop/components/JASP/Widgets/CustomMenu.qml]: Sure

**Review by JorisGoosen — APPROVED** [review]
Nice, feels very natural!

Too bad the filemenu/modulesmenu dont open when going there from an opened custommenu+ribbonbutton. But that really is more of a nice to have anyway.

**Comment by RensDofferhoff** [comment]
> Nice, feels very natural!
> 
> Too bad the filemenu/modulesmenu dont open when going there from an opened custommenu+ribbonbutton. But that really is more of a nice to have anyway.

I can make it so they always open when they get activeFocus. But tab walks you into the custom menu, so only we would need to turn that off in that case.  What is best is subjective I suppose.

**Comment by JorisGoosen** [comment]
> I make it so they always open when they get activeFocus. But tab walks you into the custom menu, so only we would need to turn that off in that case. What is best is subjective I suppose.

Hmm, Ill have to see again to understand exactly what you mean. Ill see what happens once ive got a jasp with qt 6.4.1 up and running

**Comment by RensDofferhoff** [comment]
> > I make it so they always open when they get activeFocus. But tab walks you into the custom menu, so only we would need to turn that off in that case. What is best is subjective I suppose.
> 
> Hmm, Ill have to see again to understand exactly what you mean. Ill see what happens once ive got a jasp with qt 6.4.1 up and running

We can discuss it Wednesday. As you said its nice to haves.

**Review by boutinb — APPROVED** [review]

**Pull request merged**